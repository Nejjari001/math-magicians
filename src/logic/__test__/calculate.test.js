/* eslint-disable import/order */
/* eslint-disable quotes */
import calculate from "../calculate";
import renderer from "react-test-renderer";
import Calculator from "../../components/Calculator";

describe("testing the calculate function", () => {
  it("matches the snapshot", () => {
    const tree = renderer.create(<Calculator />).toJSON();
    expect(tree).toMatchSnapshot();
  });
  const obj = {
    total: 0,
    next: null,
    operation: null,
  };

  // Addition
  test("test addition", () => {
    let result = calculate(obj, "1");
    result = calculate(result, "+");
    result = calculate(result, "2");
    result = calculate(result, "=");
    expect(result.total).toBe("3");
  });

  // Substraction
  test("test Substraction", () => {
    let result = calculate(obj, "5");
    result = calculate(result, "-");
    result = calculate(result, "3");
    result = calculate(result, "=");
    expect(result.total).toBe("2");
  });

  // Division
  test("test Division", () => {
    let result = calculate(obj, "5");
    result = calculate(result, "รท");
    result = calculate(result, "2");
    result = calculate(result, "=");
    expect(result.total).toBe("2.5");
  });

  // Mod
  test("test Mod", () => {
    let result = calculate(obj, "9");
    result = calculate(result, "%");
    result = calculate(result, "4");
    result = calculate(result, "=");
    expect(result.total).toBe("1");
  });

  // Multiplication
  test("test Multiplication", () => {
    let result = calculate(obj, "2");
    result = calculate(result, "x");
    result = calculate(result, "3");
    result = calculate(result, "=");
    expect(result.total).toBe("6");
  });
});
