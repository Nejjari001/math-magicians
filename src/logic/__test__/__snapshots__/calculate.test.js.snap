// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing the calculate function matches the snapshot 1`] = `
<div
  className="calc-container"
>
  <table
    cellSpacing="0"
    id="calc"
  >
    <tbody
      id="calc-body"
    >
      <tr
        className="result-container"
      >
        <td
          className="result"
          colSpan="4"
        >
          <span>
            0
          </span>
          <span />
          <span />
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="non-operators ac"
            onClick={[Function]}
            type="button"
          >
            AC
          </button>
        </td>
        <td>
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            +/-
          </button>
        </td>
        <td>
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            %
          </button>
        </td>
        <td>
          <button
            className="operators"
            onClick={[Function]}
            type="button"
          >
            รท
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            7
          </button>
        </td>
        <td>
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            8
          </button>
        </td>
        <td>
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            9
          </button>
        </td>
        <td>
          <button
            className="operators"
            onClick={[Function]}
            type="button"
          >
            x
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            4
          </button>
        </td>
        <td>
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            5
          </button>
        </td>
        <td>
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            6
          </button>
        </td>
        <td>
          <button
            className="operators"
            onClick={[Function]}
            type="button"
          >
            -
          </button>
        </td>
      </tr>
      <tr>
        <td>
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            1
          </button>
        </td>
        <td>
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            2
          </button>
        </td>
        <td>
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            3
          </button>
        </td>
        <td>
          <button
            className="operators"
            onClick={[Function]}
            type="button"
          >
            +
          </button>
        </td>
      </tr>
      <tr>
        <td
          colSpan="2"
        >
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            0
          </button>
        </td>
        <td>
          <button
            className="non-operators"
            onClick={[Function]}
            type="button"
          >
            .
          </button>
        </td>
        <td>
          <button
            className="operators"
            onClick={[Function]}
            type="button"
          >
            =
          </button>
        </td>
      </tr>
    </tbody>
  </table>
</div>
`;
